src/codeu/chat/server/Server.java

```diff
@@ -66,14 +73,44 @@
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
-  private static ServerInfo info;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  public Server(final Uuid id, final Secret secret, final Relay relay) throws IOException {
```

Comment:
Basically, this constructor should not have `throws IOException`. The two options I suggested above help you resolve it.